search.py

```diff
@@ -144,6 +177,27 @@ def cosine_score(tokens_arr):
 
     return results
 
+def find_term_specific_weight_for_specified_id(doc_id, posting_list):
+    """
+    Returns the accumulated tf (regardless of field type) for the stated document of doc_id seen in posting_list (which is a PostingList for a given dictionary term)
+    """
+
+    result = 0 # remains 0 if the doc_id marked relevant does not contain the term that the PostingList represents for
+    tf = 0
+
+    # scan through posting_list and accumulate to get the specified document's total tf regardless of field type
+    for posting in posting_list.postings:
+        if (posting.doc_id == doc_id):
+            # number of positions in positional index is the number of occurrences of this term in that field
+            tf += len(posting.positions)
+
+    # if the specified document does contain the term, return accumulated tf, otherwise return 0
+    if (tf > 0):
+        result = 1 + math.log(len(posting.positions), 10)
+
+    return result
+
+
```

Comment:
You aren't actually getting the document vector here right? I guess I should have mentioned this earlier, but now I see you might have to make changes to `index.py`.

Reply:
By doing this your document vector only has non zero values for terms that are found in the query, which isnt a proper representation of the document vector

Reply:
Not saying that theres an easy solution

Reply:
What this does is to go through the postingList, find all the zones' Postings for a single document, then add their tf before calculating their contribution towards shifting the initial query vector. 

This works because we process term by term, so we only change the query vector's entry for this current term we are focusing on. (which we have just calculated the contribution value here, which needs to be accumulated for all the relevant documents, then averaged, before actually shifting the initial query's value for the current term.)